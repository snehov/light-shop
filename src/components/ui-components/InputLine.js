import React from 'react'
import Input from './Inputjs.js'

const InputLine = ({ name, groupName = '', type, label, ...props }) => {
  const id = groupName ? `${groupName}_${name}` : name

  return (
    <div className="inputLine">
      <label className="labelToInput" htmlFor={id}>
        {label}
      </label>
      <Input name={name} type={type} id={id} {...props} />
    </div>
  )
}
export default InputLine
