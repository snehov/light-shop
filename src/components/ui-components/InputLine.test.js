import React from 'react';
import InputLine from './InputLine';
import renderer from 'react-test-renderer';

test('Test input line',()=>{
  const component = renderer.create(
    <InputLine name="testInput" type="text" label="testing"/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})