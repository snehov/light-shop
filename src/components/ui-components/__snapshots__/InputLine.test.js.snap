// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test input line 1`] = `
<div
  className="inputLine"
>
  <label
    htmlFor="testInput"
  >
    testing
  </label>
  <span>
    <input
      className=""
      id="testInput"
      name="testInput"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </span>
</div>
`;

exports[`Test input line: 
  <div
    className="inputLine"
  >
    <label
      htmlFor="testInput"
    >
      testing
    </label>
    <span>
      <input
        className=""
        id="testInput"
        name="testInput"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
    </span>
  </div>
   1`] = `
<div
  className="inputLine"
>
  <label
    htmlFor="testInput"
  >
    testing
  </label>
  <span>
    <input
      className=""
      id="testInput"
      name="testInput"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </span>
</div>
`;
